src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -382,6 +391,109 @@ private void setAssignedTeamIdForExerciseAndUser(Exercise exercise, User user) {
         }
     }
 
+    /**
+     * Get specific exercise information
+     *
+     * @param courseId the id of the course
+     * @return An Integer array containing active students for each index
+     */
+    public List<CourseManagementOverviewExerciseDetailsDTO> getExercisesForCourseManagementOverview(Long courseId) {
+        List<CourseManagementOverviewExerciseDetailsDTO> detailsDTOS = new ArrayList<>();
+        for (var listElement : exerciseRepository.getExercisesForCourseManagementOverview(courseId)) {
+            var exerciseId = (Long) listElement.get("id");
+            var exerciseType = listElement.get("type");
+            if (exerciseId == null || exerciseId == 0 || exerciseType == null) {
+                continue;
+            }
+
+            var dto = new CourseManagementOverviewExerciseDetailsDTO();
+            dto.setExerciseId(exerciseId);
+            dto.setExerciseTitle((String) listElement.get("title"));
+
+            if (exerciseType == QuizExercise.class) {
```

Comment:
Seems to be the perfect place for a switch maybe?

Reply:
The solution isn't very nice, but the type is not switchable, so we can't use that.